app/views/posts/index.html.erb

```diff
@@ -1,4 +1,4 @@
-<%= link_to new_post_path, id: 'new_post' do %>
+<%= link_to "#{current_user.id}/posts/new", id: 'new_post' do %>
```

Comment:
I'd be interested to understand why not new_post_path?

Reply:
new_post_path goes to /posts/new, but we need a way to remember which wall we are going to post to.
Methods in order:
show_wall
new
create
- #new needs to remember the wall id from show_wall to pass it to create.